crates/swc_ecma_ast/src/typescript.rs

```diff
@@ -546,6 +546,8 @@ pub struct TsImportType {
     pub qualifier: Option<TsEntityName>,
     #[cfg_attr(feature = "serde-impl", serde(rename = "typeArguments"))]
     pub type_args: Option<Box<TsTypeParamInstantiation>>,
+    #[cfg_attr(feature = "serde-impl", serde(default))]
+    pub with: Option<Box<ObjectLit>>,
```

Comment:
I'm not sure about this name because this is the full `{ with: { "module-resolution": "import" } }` (which I need for dprint-plugin-typescript). Maybe something else would be better here?

Reply:
I'll wait to update the failing tests until getting feedback on this.

Reply:
So `annotation`? I think we may need to use separate type for this kind of annotations. How do you think? cc @swc-project/core-es

Reply:
Is `with` hardcoded?
I hope there is no situation like `{ with: { "foo": "bar" }, other: "value" }`.

Reply:
> The expected type of that second argument is defined by a type called `ImportCallOptions`, which by default just expects a property called `with`.

And typescript playground says `Object literal may only specify known properties, and 'xxx' does not exist in type 'ImportCallOptions'.`

So I think some names related to `ImportCallOptions` are ok as well.

Reply:
I think we need a breaking change anyway if there's a situation like `{ with: { "foo": "bar" }, other: "value" }`